base/compiler/typeinfer.jl

```diff
@@ -381,6 +384,64 @@ function cache_result!(interp::AbstractInterpreter, result::InferenceResult)
     nothing
 end
 
+function finish_opaque_closure!(clos::PartialOpaque, mod::Module, interp::AbstractInterpreter)
+    # Infer with the most general types possible, so that optimization has
+    # access to the result.
```

Comment:
The implication in `finish` is that inference is already done, since otherwise we might still have an unresolved cycle that must be converged first before we can annotate the correct (sufficiently wide) types. Thus, this is not a point in the graph where it is valid to look to look at IPO information.